core/src/banking_stage.rs

```diff
@@ -1286,6 +1260,12 @@ impl BankingStage {
             let commit_time_us = commit_time.as_us();
             execute_and_commit_timings.commit_us = commit_time_us;
 
+            let commit_transaction_statuses = tx_results
+                .execution_results
+                .iter()
+                .map(|execution_result| execution_result.was_executed())
+                .collect();
+
```

Comment:
This moves the commit_transaction_statuses from after `record` to after `bank.commit`, not 100% sure the difference between that two stages, but probably makes more sense to undo cost tracker if transactions are not committed to bank.

Reply:
If record fails, `commit_transactions_result` will be an error and qos will remove all costs